cpp/csp/python/PyStructToJson.cpp

```diff
@@ -198,12 +198,38 @@ rapidjson::Value toJsonRecursive( const StructPtr& self, rapidjson::Document& do
     return new_dict;
 }
 
-rapidjson::Value pyDictKeyToName( PyObject * py_key, rapidjson::Document& doc )
+std::pair<const char *, Py_ssize_t> pyObjectToString( PyObject * py_obj )
 {
-    // Only support strings, ints, and floats as keys
+    auto * str_obj = PyObject_Str( py_obj );
+    Py_ssize_t len = 0;
+    const char * str = PyUnicode_AsUTF8AndSize( str_obj, &len );
+    return std::make_pair( str, len );
+}
+
+rapidjson::Value pyDictKeyToName( PyObject * py_key, rapidjson::Document& doc, PyObject * callable )
+{
+    // NOTE: Only support None, bool, strings, ints, and floats, date, time, datetime, enums, csp.Enums as keys
     // JSON encoding requires all names to be strings so convert them to strings
+
+    static thread_local PyTypeObjectPtr s_tl_enum_type;
+    // Get the enum type on the first call and save it for future use
+    if( s_tl_enum_type.get() == nullptr )
+    {
+        // Import enum module to extract the Enum type
+        auto py_enum_module = PyObjectPtr::own( PyImport_ImportModule( "enum" ) );
+        if( py_enum_module.get() )
+        {
+            s_tl_enum_type = PyTypeObjectPtr::own( reinterpret_cast<PyTypeObject*>( PyObject_GetAttrString( py_enum_module.get(), "Enum" ) ) );
+        }
```

Comment:
If the import of `enum` fails for some reason, this code block will be invoked every time `pyDictKeyToName` is called.
Since `enum` is part of the Python standard library I really don't think the import could fail in the first place; if it does we should just throw because something is very wrong.